torch/csrc/autograd/input_metadata.h

```diff
@@ -67,9 +105,33 @@ struct InputMetadata {
     return at::zeros(shape_, options_);
   }
 
+  at::IntArrayRef compute_shape(const at::Tensor& t){
+    if(t.is_nested()){
+      return at::IntArrayRef{};
```

Comment:
So this is used for the constructor when metadata is passed in a tensor. The shape field on metadata should store the smallest amount of data when it is nested. But maybe I should be actually be returning: `DimVector{};` otherwise just grab the sizes from the tensor.